impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -245,12 +244,75 @@ public Map<String, Object> getKafkaClientConfiguration(String user, String passw
         return clusterConfig.getConnectConfigForCluster(getBootstrapServers(), user, password);
     }
 
+    @Override
+    public synchronized int addBroker() {
+        // find next free kafka node.id
+        var first = IntStream.rangeClosed(0, getNumOfBrokers()).filter(cand -> !servers.containsKey(cand)).findFirst();
+        if (first.isEmpty()) {
+            throw new IllegalStateException("Could not determine new nodeId, existing set " + servers.keySet());
+        }
+        var newNodeId = first.getAsInt();
+
+        LOGGER.log(System.Logger.Level.DEBUG,
+                "Adding broker with node.id {0} to cluster with existing nodes {1}.", newNodeId, servers.keySet());
+
+        // preallocate ports for the new broker
+        try (var preallocator = new ListeningSocketPreallocator()) {
+            ports.get(Listener.EXTERNAL).put(newNodeId, preallocator.preAllocateListeningSockets(1).get(0));
+            ports.get(Listener.ANON).put(newNodeId, preallocator.preAllocateListeningSockets(1).get(0));
+            ports.get(Listener.INTERNAL).put(newNodeId, preallocator.preAllocateListeningSockets(1).get(0));
+        }
+
+        var configHolder = clusterConfig.generateConfigForSpecificBroker(this, newNodeId);
+        final Server server = buildKafkaServer(configHolder);
+        tryToStartServerWithRetry(configHolder, server);
+        servers.put(configHolder.getBrokerNum(), server);
```

Comment:
I think we should really use `newNodeId` here or change `configHolder.getBrokerNum` to `configHolder.getBrokerId`

Reply:
I think `node.id` is the new parlance.  I think Kafka adopted it because with KRaft nodes aren't necessarily in the `broker` role, so `broker.id` would have been confusing.  I think it makes sense for us to adopt the new term.